ush/python/pygfs/task/stage_ic.py

```diff
@@ -1,58 +1,481 @@
+            List of member indices to process
+        """
+        if run in ['enkfgdas']:
+            return list(range(1, nmem_ens + 1))
+        elif run in ['gefs']:
+            if gefstype == 'gefs-real-time':
+                # Map GEFS members to GFS member numbers
+                # Cycle ranges determine which 30-member range to use based on cycle index
+                cyc_ranges = {'00': list(range(1, 31)),
+                              '06': list(range(21, 51)),
+                              '12': list(range(41, 71)),
+                              '18': list(range(61, 81)) + list(range(1, 11))}
+                member_list = [0] + cyc_ranges[f'{m_index:02}']
+                return member_list
+            else:
+                # GEFS offline uses sequential member numbering
+                return list(range(0, nmem_ens + 1))
+        else:
+            # Deterministic runs (GFS, GDAS, GCAFS, GCDAS)
+            # Only member -1 is processed (empty string for MEMDIR)
+            return [-1]
+
+    @staticmethod
+    def _paths_from_templates(stage_dict: AttrDict, com_path_tuples: Tuple[Tuple[str, str, Dict[str, Any]], ...]) -> Dict[str, str]:
+        """Generate COM paths from template configurations
+
+        Parameters
+        ----------
+        stage_dict : AttrDict
+            Staging configuration dictionary
+        com_path_tuples : Tuple[Tuple[str, str, Dict[str, Any]], ...]
+            Tuple of tuples containing path key, template key, and substitution dict
+
+        Returns
+        -------
+        Dict[str, str]
+            Dictionary mapping COM path keys to resolved file paths
+        """
+        def _replace_template_vars(template: str, var_dict: Dict[str, Any]) -> str:
+            """Replace template variables in string with actual values
+
+            Parameters
+            ----------
+            template : str
+                Template string with variables to replace
+            var_dict : Dict[str, Any]
+                Dictionary of variable names and values
+
+            Returns
+            -------
+            str
+                String with variables replaced
+            """
+            replaced_com = template
+            for var, value in var_dict.items():
+                replaced_com = replaced_com.replace(var, value)
+            return replaced_com
+
+        path_dict = {}
+        for com_key, template_key, substitution_dict in com_path_tuples:
+            template_str = stage_dict.get(template_key, '')
+            if not template_str:
+                logger.warning("Template key '%s' not found for COM key '%s'", template_key, com_key)
+                path_dict[com_key] = ''
+            else:
+                path_dict[com_key] = _replace_template_vars(template_str, substitution_dict)
+        return path_dict
+
+    @logit(logger)
+    def _get_member_com_paths_gfs(self, stage_dict: AttrDict, member: int) -> Tuple[Tuple[str, str, Dict[str, Any]], ...]:
```

Comment:
```suggestion
   @staticmethod
    def _get_member_com_paths_gfs( stage_dict: AttrDict, member: int) -> Tuple[Tuple[str, str, Dict[str, Any]], ...]:
```
These and the following are staticmethods, they don't need `self`.